src/inventory/gilded_rose.js

```diff
@@ -21,59 +21,99 @@ const items = [
+      case sell_in > 10: 
+        increaseQuality(item, 1);
+        break;
+      // //increase the quality by 2 if the sell_in is between 6-10
+      case sell_in > 5 && sell_in <= 10:
+        increaseQuality(item, 2);
+        break;
+      // //increase the quality by 3 if the sell_in is between 1-5
+      case sell_in > 0 && sell_in <= 5:
+        increaseQuality(item, 3);
+        break;
+      //decrease quality all the way if the sell_in is 0 or less
+      case sell_in <= 0: 
+        item.quality = item.quality - item.quality;
+        break;
+    }
+  }
+}
+
 export function updateQuality(items) {
   for (var i = 0; i < items.length; i++) {
-    // if it's anything but the aged brie, the passes, sulfuras, && the quality is still above 0, reduce quality by 1
-    if (items[i].name != 'Aged Brie' && 
-        items[i].name != 'Backstage passes to a TAFKAL80ETC concert' && 
-        items[i].name != 'Sulfuras, Hand of Ragnaros' && 
-        items[i].quality > 0) {
-          items[i].quality = items[i].quality - 1;
-    } else { //OTHERWISE...
-        // if the quality is still under 50, && the item is either the brie or the passes, increase the quality by 1
-        if ((items[i].quality < 50) &&
-            (items[i].name == 'Backstage passes to a TAFKAL80ETC concert' || 
-            items[i].name == 'Aged Brie')) {
-              items[i].quality = items[i].quality + 1
-          }
-        // ALSO...if the item is ONLY the passes and the quality is still less than 50...
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert' && 
-            items[i].quality < 50) {
-          // and the sell_in this less than 11, increase the quality AGAIN by 1 (2 total)
-          if (items[i].sell_in <= 10) {
-            items[i].quality = items[i].quality + 1
-          }
-          // and the sell_in this less than 6, increase the quality AGAIN by 1 (3 total)
-          if (items[i].sell_in <= 5) {
-            items[i].quality = items[i].quality + 1
-          }
-          // and once the passes get to sell_in of 0, the lose all quality
-          if (items[i].sell_in <= 0) {
-            items[i].quality = items[i].quality - items[i].quality
-          }
-        }
-      }
-    
-    // ALSO...if the item is not sulfuras, the sell_in will go down by 1
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
+    // for "standard" items
+    if (checkNotSpecialItem(items[i])) {
+      // if it's anything but the aged brie, the passes, sulfuras, first reduce the sell_in by 1
+      decreaseSellIn(items[i]);
+      // checks the quality and sell_in for "standard" items and changes values accordingly
+      standardItemChanges(items[i]);
+      // continue with the next item in the loop
+      continue;
+    }
+    // if the item is sulfuras, don't do anything, and continue to the next item in the loop
+    if (items[i].name === 'Sulfuras, Hand of Ragnaros') {
```

Comment:
Minor but could be shortened to one line
```suggestion
    if (items[i].name === 'Sulfuras, Hand of Ragnaros') continue;
```

Reply:
✅ didn't know I could do that